lib/paragraphs.rb

```diff
@@ -0,0 +1,25 @@
+class Para
+
+  def paragraphs(input)
+    input.each_with_index.map do |x, idx|
+      if x.include?("\n") == true || x.include?("<h") == true || x.include?("<li>")
+        x
+      elsif input[idx-1] == "\n\n" && input[idx+1] == "\n\n"
+        x.insert(0, "<p>"); x.insert(-1, "</p>")
+      elsif input[idx-1] == "\n\n" && input[idx+1] == "\n"
+        x.insert(0, "<p>")
+      elsif input[idx-1] == "\n" && input[idx+1] == "\n\n"
```

Comment:
what is `input[idx - 1]`

this can be a method since it is being used so heavily